application/org.openjdk.jmc.console.twitter/src/main/java/org/openjdk/jmc/console/twitter/TwitterOAuthAunthenticator.java

```diff
@@ -0,0 +1,242 @@
+
+			String authorization_header_string = "OAuth oauth_consumer_key=\"" + consumerKey
+					+ "\",oauth_signature_method=\"" + OAUTH_SIGNATURE_METHOD + "\",oauth_timestamp=\""
+					+ oauth_timestamp + "\",oauth_token=\"" + oauth_token + "\",oauth_verifier=\"" + oauth_verifier
+					+ "\",oauth_nonce=\"" + oauth_nonce + "\",oauth_version=\"1.0\",oauth_signature=\""
+					+ URLEncoder.encode(oauth_signature, "UTF-8") + "\"";
+
+			HttpRequest request = HttpRequest.newBuilder().POST(ofFormData(new HashMap<String, String>()))
+					.uri(URI.create(OAUTH_ACCESS_TOKEN_URL)).setHeader("Authorization", authorization_header_string)
+					.header("Content-Type", "application/x-www-form-urlencoded").build();
+
+			HttpResponse<String> response = getHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+			String responseBody = response.body();
+			AccessToken accessToken = null;
+			if (response.statusCode() == 200) {
+				String final_oauth_token = responseBody.substring(responseBody.indexOf("oauth_token=") + 12,
+						responseBody.indexOf("&oauth_token_secret="));
+				String final_oauth_token_secret = responseBody
+						.substring(responseBody.indexOf("oauth_token_secret=") + 19, responseBody.indexOf("&user_id="));
+				String userName = responseBody.substring(responseBody.indexOf("screen_name=") + 12,
+						responseBody.length());
+				accessToken = new AccessToken(final_oauth_token, final_oauth_token_secret, userName);
+			}
+
+			return accessToken;
+		} catch (Exception e) {
+			LOGGER.log(Level.SEVERE, "Unable to authenticate.", e);
+		}
+		return null;
+
+	}
+
+	private static void configureProxySettings() {
+		System.setProperty("java.net.useSystemProxies", "true");
+		Proxy proxy;
+		try {
+			proxy = (Proxy) ProxySelector.getDefault().select(new URI("http://api.twitter.com")).iterator().next();
+
+			InetSocketAddress addr = (InetSocketAddress) proxy.address();
+			if (addr != null) {
+				if (addr.getHostName() != null) {
+					proxyHost = addr.getHostName();
+				}
+				if (addr.getPort() != -1) {
+					proxyPort = addr.getPort();
+				} else {
+					proxyPort = 80;
+				}
+			}
+		} catch (URISyntaxException e) {
+			// Should never happen...
+			LOGGER.log(Level.SEVERE, "Failed to parse URI", e);
+		}
+	}
+
+	private static HttpClient getHttpClient() {
+		configureProxySettings();
+		InetSocketAddress addr;
+		if (proxyHost != null) {
+			addr = new InetSocketAddress(proxyHost, proxyPort);
+		} else {
+			addr = null;
+		}
+		HttpClient httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_2)
+				.connectTimeout(Duration.ofSeconds(100)).proxy(ProxySelector.of(addr)).build();
+
+		return httpClient;
+	}
+
+	private static String computeSignature(String baseString, String keyString) throws Exception {
+
+		SecretKey secretKey = null;
+
+		byte[] keyBytes = keyString.getBytes();
+		secretKey = new SecretKeySpec(keyBytes, "HmacSHA1");
```

Comment:
Perhaps the "HmacSHA1" could be a global variable instead of a quoted string here (as it's used again below)?